### ISSUE TITLE
`indi-libcamera` fails to build in version 2.0.5

### ISSUE THREAD

**Issue by hamarituc** [indilib/indi-3rdparty#860]
**Describe the bug**

`indi-libcamera` fails to build

**To Reproduce**

1. Compile `indi-libcamera`
2. See error

```
FAILED: CMakeFiles/indi_libcamera_ccd.dir/indi_libcamera.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DLIBAV_PRESENT=0 -DLIBDRM_PRESENT=0 -DLIBEGL_PRESENT=0 -DOPENCV_PRESENT=1 -DQT_PRESENT=0 -I/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera_build -I/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera -I/usr/include/libindi -I/usr/include/libraw -I/usr/include/libcamera -I/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/libcamera-apps  -march=native -O2 -pipe -D_FORTIFY_SOURCE=2 -fstack-protector-all -fPIE -O1 -Wa,--noexecstack  -Wall -Wextra -Wno-unused-but-set-variable -Wno-format-truncation -g -std=gnu++17 -fPIE -MD -MT CMakeFiles/indi_libcamera_ccd.dir/indi_libcamera.cpp.o -MF CMakeFiles/indi_libcamera_ccd.dir/indi_libcamera.cpp.o.d -o CMakeFiles/indi_libcamera_ccd.dir/indi_libcamera.cpp.o -c /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp
In file included from /usr/include/libindi/defaultdevice.h:23,
                 from /usr/include/libindi/indiccd.h:30,
                 from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.h:34,
                 from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:21:
/usr/include/libindi/indilogger.h:72: warning: "LOG_ERROR" redefined
   72 | #define LOG_ERROR(txt)  DEBUG(INDI::Logger::DBG_ERROR, (txt))
      | 
In file included from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/libcamera-apps/core/options.hpp:21,
                 from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/libcamera-apps/core/video_options.hpp:14,
                 from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/libcamera-apps/core/libcamera_encoder.hpp:10,
                 from /var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.h:29:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/libcamera-apps/core/logging.hpp:9: note: this is the location of the previous definition
    9 | #define LOG_ERROR(text) std::cerr << text << std::endl
      | 
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘void INDILibCamera::initSwitch(INDI::PropertySwitch&, int, const char**)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:475:26: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  475 |     for (size_t i = 0; i < n; i++)
      |                        ~~^~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘void INDILibCamera::initOptions(bool)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:634:38: warning: unused parameter ‘video’ [-Wunused-parameter]
  634 | void INDILibCamera::initOptions(bool video)
      |                                 ~~~~~^~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘bool INDILibCamera::processRAW(const char*, uint8_t**, size_t*, int*, int*, int*, int*, char*)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1053:39: error: ‘IDSharedBlobRealloc’ was not declared in this scope
 1053 |     *memptr  = static_cast<uint8_t *>(IDSharedBlobRealloc(*memptr, *memsize));
      |                                       ^~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1055:42: error: ‘IDSharedBlobAlloc’ was not declared in this scope
 1055 |         *memptr = static_cast<uint8_t *>(IDSharedBlobAlloc(*memsize));
      |                                          ^~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘bool INDILibCamera::processJPEG(const char*, uint8_t**, size_t*, int*, int*, int*)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1112:39: error: ‘IDSharedBlobRealloc’ was not declared in this scope
 1112 |     *memptr  = static_cast<uint8_t *>(IDSharedBlobRealloc(*memptr, *memsize));
      |                                       ^~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1114:42: error: ‘IDSharedBlobAlloc’ was not declared in this scope
 1114 |         *memptr = static_cast<uint8_t *>(IDSharedBlobAlloc(*memsize));
      |                                          ^~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘int INDILibCamera::processJPEGMemory(unsigned char*, long unsigned int, uint8_t**, size_t*, int*, int*, int*)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1197:39: error: ‘IDSharedBlobRealloc’ was not declared in this scope
 1197 |     *memptr  = static_cast<uint8_t *>(IDSharedBlobRealloc(*memptr, *memsize));
      |                                       ^~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:1199:42: error: ‘IDSharedBlobAlloc’ was not declared in this scope
 1199 |         *memptr = static_cast<uint8_t *>(IDSharedBlobAlloc(*memsize));
      |                                          ^~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp: In member function ‘void INDILibCamera::workerExposure(const std::atomic_bool&, float)’:
/var/tmp/portage/sci-libs/indilib-driver-libcamera-2.0.5/work/indi-3rdparty-2.0.5/indi-libcamera/indi_libcamera.cpp:390:21: warning: ignoring return value of ‘ssize_t read(int, void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  390 |                 read(fd, memptr, memsize);
      |                 ~~~~^~~~~~~~~~~~~~~~~~~~~
```

The functions `IDSharedBlobRealloc()` and `IDSharedBlobAlloc()` are not declared.

**Expected behavior**

The driver compiles.

**Screenshots**

N/A

**Desktop (please complete the following information):**

- Linux
- Version 2.0.5

**Log Files**

N/A

**Comment by knro** [comment]
Should be fixed by b0e7f51ae996e19a772940dd0b0c9d4e6d9333d3

**Comment by hamarituc** [comment]
Sorry for the noise. I missed to check the commits since the last release. With the patch, the drivers compiles now. Thanks.
